src/native/fs/fs_napi.cpp

```diff
@@ -30,13 +32,68 @@
     #include <sys/statfs.h>
 #endif
 
+// #define GPFS_ATTRFLAG_INCL_ENCR          0x0040  /* Include encryption attributes */
+// #define GPFS_DOT_ENCRYPTION_EA "Encryption"
+
+// typedef struct gpfs_fssnap_handle gpfs_fssnap_handle_t;
+// typedef struct gpfs_ifile gpfs_ifile_t;
+// typedef unsigned int gpfs_ino_t;
+// typedef struct gpfs_iattr gpfs_iattr_t;
+
+// typedef int gpfs_file_t;
+// typedef struct gpfsFcntlHeader gpfsFcntlHeader_t;
+// typedef struct gpfsGetSetXAttr gpfsGetSetXAttr_t;
+
+#define ROUNDUP(X,Y) ( (Y) * ( ((X)+(Y)-1) / (Y) ) )
+
+#define GPFS_DOT_ENCRYPTION_EA "Encryption"
+#define GPFS_ATTR_INTERNAL_APPL_ID   3
+#define GPFS_ATTR_EXT_NAMESPACE_ID   4
+
+#define XATTR_APPID_TO_NAMESPACE(id) \
+  (id == GPFS_ATTR_INTERNAL_APPL_ID) ? "gpfs" : ""
+#define XATTR_APPID_TO_NAMESPACE_LEN(id) \
+  (id == GPFS_ATTR_INTERNAL_APPL_ID) ? sizeof("gpfs")-1 : 0
+
+#define GPFS_FCNTL_CURRENT_VERSION     1
+#define GPFS_FCNTL_GET_XATTR           6001
+#define GPFS_FCNTL_XATTRFLAG_NONE      0x0000
+
+#define GPFS_FCNTL_ERR_NO_ATTR             27
+#define GPFS_FCNTL_ERR_ATTR_EXISTS         28
+#define GPFS_FCNTL_ERR_BUFFER_TOO_SMALL    29
+#define GPFS_FCNTL_ERR_NO_ATTR_SPACE       30
+
 namespace noobaa
 {
 
 DBG_INIT(0);
 
+gpfs_fssnap_handle_t *(*gpfs_get_fssnaphandle_by_path)(const char*);
+
+gpfs_ifile_t *(*gpfs_iopen)(gpfs_fssnap_handle_t *fssnapHandle,
+    gpfs_ino_t ino,
+    int open_flags,
+    const gpfs_iattr_t *statxbuf,
+    const char *symLink);
+
+int (*gpfs_igetattrsx)(gpfs_ifile_t *ifile,
+    int flags,
+       void *buffer,
+       int bufferSize,
+       int *attrSize);
+
+int (*gpfs_fcntl)(gpfs_file_t fileDesc,
+           void* fcntlArgP);
+
 static int (*dlsym_gpfs_fcntl)(gpfs_file_t file, void* arg) = 0;
 
+int (*gpfs_linkat)(gpfs_file_t fileDesc, const char *oldpath,
```

Comment:
probably also old code, needs to be using only one of those and probably the static ones like we did in gpfs_fcntl

Reply:
is needed, and used in tempFile

Reply:
I think you should be using only this one: dlsym_gpfs_linkat (line 41)
As discussed in my previous PR, we should work with the static version (I think it's again mixes of old and new versions - before the fixes I made to my PR and after)

Reply:
will fix

Reply:
can you resolve this as well ^

Reply:
done